File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pingyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

public interface BrandService {
    public List<TbBrand> findAll();


    //分页的方法
    public PageResult findPage(int pageNum,int size);

    void add(TbBrand tbBrand);

    void delete(Long[] ids);

    PageResult search(int page, int rows, TbBrand tbBrand);

    TbBrand findOne(Long id);

    void update(TbBrand tbBrand);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//自定义服务实现然后实现service层和controller层的代码重用
app.service("brandService", function ($http) {
    this.searchBrand = function (page, rows, eleAaa) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, eleAaa);
    }

    this.save = function (methodName,eleAaa) {
        return $http.post("../brand/"+methodName+".do", eleAaa);
    }

    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }

    this.findOne = function (id) {
        return $http.post("../brand/findOne.do?id=" + id);
    }

    this.selectOptionList=function () {
        return $http.post("../brand/selectOptionList.do");
    }
})